loris/resolver.py

```diff
@@ -61,13 +65,40 @@ def resolve(self, ident):
             ident (str):
                 The identifier for the image.
         Returns:
-            (str, str): (fp, format)
+            ImageInfo: Partially constructed ImageInfo object 
```

Comment:
The partially constructed object is still useful for some operations, notably testing whether the request is authorized which needs a lot of pointers but not the actual pixel information.  The other case is the info cache, but there it's constructed via from_json(). There are a few tests that occur between constructing the info object and loading pixels if needed, which are different in get_info and get_img.

As to the suggestions, yes, they could work. My opinions:

* Returning an array of variables rather than an object is (IMO) an anti-pattern in any object oriented paradigm. We'll always run into situations where we need just one more piece of information that could easily (and probably already is) on the info object. I can see the point of @jpstroop's tuple, but once you start needing more, my experience is that it's a very slippery slope into needing everything.
* I worry about overloading the resolve() to do lots of testing (per get_info and get_img) that need application level configuration information. That also seems like a slope towards a single function that does everything (which I did in my first implementation of the API, and was terrible).

`from_image_file()` returns self, so there's a reasonably convenient pattern if you have all the information and just need the whole thing:

`info = ImageInfo(ident, fp).from_image_file()`